### PR TITLE
[Emulator tests] Add proxy server-4

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
- Follow up PR on https://github.com/GoogleCloudPlatform/gcsfuse/pull/2738
- Added emulator.go file to generate a retry ID and attach it to the request header.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Updated
3. Integration tests - NA

### CODE REVIEW

File: tools/integration_tests/emulator_tests/proxy_server/emulator.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	endpoint, err := url.Parse(host)
+	if err != nil {
+		return "", fmt.Errorf("Failed to parse host env: %v\n", err)
+	}
+
+	et := &emulatorTest{host: endpoint}
+	return et.GetRetryID(instructions, "http")
+}
+
+func AddRetryID(req *http.Request, r RequestTypeAndInstruction) error {
```
Comment (raj-prince) [inline tools/integration_tests/emulator_tests/proxy_server/emulator.go:88]: Let's keep this method in the main proxy server code.

File: tools/integration_tests/emulator_tests/proxy_server/emulator.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	}{}
+	if err := json.NewDecoder(resp.Body).Decode(&testRes); err != nil {
+		return "", fmt.Errorf("decoding test ID: %v\n", err)
+	}
+
+	et.host.Path = ""
+	return testRes.TestID, nil
+}
+
+// CreateRetryTest creates a retry test using the provided instructions.
```
Comment (raj-prince) [inline tools/integration_tests/emulator_tests/proxy_server/emulator.go:73]: Could you please elaborate the comment, specially about the return type?

File: tools/integration_tests/emulator_tests/proxy_server/emulator.go (hunk 1 above)
Comment (raj-prince) [reply tools/integration_tests/emulator_tests/proxy_server/emulator.go:88]: Also, write a doc comment on this.

Comment (Tulsishah) [reply tools/integration_tests/emulator_tests/proxy_server/emulator.go]: Done

File: tools/integration_tests/emulator_tests/proxy_server/emulator.go (hunk 2 above)
Comment (Tulsishah) [reply tools/integration_tests/emulator_tests/proxy_server/emulator.go:73]: Done

**Review by raj-prince — APPROVED** [review]

**Pull request merged**